### ISSUE TITLE
[RFC] Scripts under `.git/hooks/` may contain calls to undefined commands

### ISSUE THREAD

**Issue by phansys** [gushphp/gush#617]
I have a `prepare-commit-msg` hook, which has `#!/bin/bash` as shebang. Obviously, as the container image doesn't includes `bash`, it doesn't work and the merge commit is aborted.
Even if in my case, the situation could be resolved adding `bash` to container dependencies, I think there could be other corner cases where commit hooks invokes commands that aren't present at the containerized gush environment.
How we could proceed in front of these scenarios? I can imagine just a possible solution: Check if there is any of the hooks enabled, delete them and warn the user about that limitation.

**Comment by sstok** [comment]
Hmm, hooks should not be executed when using Gush as local operations are only done to bypass some API limitations. Is there no command option in Gush to skip/disable this for the operation?

**Comment by phansys** [comment]
@sstok, there are some options such `--no-verify` for [`commit`](https://git-scm.com/docs/git-commit#_options), which bypass the `pre-commit` and `commit-msg`. Additionally, all hooks can be disabled removing the executable bit:
>Hooks that don’t have the executable bit set are ignored.

Ref: https://git-scm.com/docs/githooks#_description

**Comment by cordoval** [comment]
yeah the problem is with the commands that are not in there, but this could include any tools and be any commands so i don't think you can include satisfactorily all cases here. The only gush commands that commit are when rebasing or a minimum set. I think you should use git on local for using your hooks and then just use gush for the main api stuff.

**Comment by phansys** [comment]
@cordoval , the RFC is about how to address the situation. I think executing `chmod -x $GIT_DIR/hooks/*` and later restore the original permissions could be a good solution in order to cover all the possibilities.

**Comment by cordoval** [comment]
oh you mean because the hooks are not seen as executables? is that what you mean?

**Comment by phansys** [comment]
No @cordoval, I mean we should disable them at gush execution time (temporarily) with some mechanism that applies for all of them. I'm opening the game here in order to hear thoughts about how to achieve that goal.

**Comment by cordoval** [comment]
i think we should include bash to the image, are you sure it does not contain sh?

**Comment by phansys** [comment]
Yes @cordoval, it contains `sh` but not `bash`. But I think adding `bash` to container dependencies is just a workaround for my specific scenario, other users can have any other dependencies on their hooks. BTW, I agree that majority of hooks just relies on simple `sh` or `bash` commands, but we should prevent the "border" cases too.
